wire-grpc-client/src/main/java/com/squareup/wire/GrpcClient.kt

```diff
@@ -37,11 +38,24 @@ class GrpcClient private constructor(
 
     return Proxy.newProxyInstance(
         service.java.classLoader,
-        arrayOf<Class<*>>(service.java)
-    ) { _, method, args ->
-      val grpcMethod = methodToService[method] as GrpcMethod<*, *>
-      grpcMethod.invoke(this@GrpcClient, args ?: emptyArray())
-    } as T
+        arrayOf<Class<*>>(service.java),
+        object : InvocationHandler {
+          override fun invoke(proxy: Any, method: Method, args: Array<Any>?): Any {
+            val self: InvocationHandler = this
+
+            if (method.declaringClass == Object::class.java) {
+              return if (args == null) {
+                method.invoke(self)
+              } else {
+                method.invoke(self, *args)
+              }
+            }
+
+            val grpcMethod = methodToService[method] as GrpcMethod<*, *>
+            return grpcMethod.invoke(this@GrpcClient, args ?: emptyArray())
```

Comment:
How about moving this `args ?: emptyArray` before the `if` to avoid doing it in both branches?

Reply:
why not, done